app/models/project/queue.rb

```diff
@@ -0,0 +1,84 @@
+# Abstract class for Project task queues.
+# Gives the logged in User a new InfoRequest to contribute to.
+#
+# Subclasses must implement #info_requests.
+class Project::Queue
+  def initialize(project, current_user, session)
+    @project = project
+    @current_user = current_user
```

Comment:
Just noticed I don't actually use `current_user` after some refactoring, so I'll drop this.